src/http-adapter.js

```diff
@@ -112,36 +123,30 @@ class HttpAdapter {
     return promise
   }
 
-  _makeRequest ({ url, type, headers, data }) {
-    return new Promise((resolve, reject) => {
-      let request = {
-        url,
-        type,
-        headers,
-        success: (data, textStatus, jqXhr) => {
-          if (!data && jqXhr.status !== 204) {
-            throw new Error(`request returned ${jqXhr.status} status without data`)
-          }
-          return resolve(data)
-        },
-        error: (response) => {
-          if (response.readyState === 0 || response.status === 0) {
-            // this is a canceled request, so we literally should do nothing
-            return
-          }
-
-          const error = new Error(`request for resource, ${url}, returned ${response.status} ${response.statusText}`)
-          error.response = response
-          reject(error)
-        },
-        // being explicit about data type so jQuery doesn't "intelligent guess" wrong
-        // changing this may not break tests, but does behave badly in prod
-        dataType: 'text',
-        data
-      }
-
-      ajax(request)
-    })
+  async _fetch ({ url, init }) {
+    let response = await fetch(url, init)
+    switch (response.status) {
+      case 200:
+      case 201:
+      case 202:
+      case 203:
+      case 205:
+      case 206:
+        let data = await response.json()
+        return data
+      case 204:
+        if (init.method == METHOD.DELETE) {
+          return {}
+        } else {
+          throw new Error(
+            `request returned ${response.status} status without data`
+          )
```

Comment:
You've inverted this conditional; it previously checked that status was _not_ 204

Reply:
204 with no data _should_ be the expected outcome. No data on a non-204 but 2xx should throw.

Reply:
ahhhhh yeah I completely mixed that up. let me change it